src/Shared/ServerInfrastructure/Http2/ConnectionEndReason.cs

```diff
@@ -0,0 +1,58 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Core;
+
+internal enum ConnectionEndReason
+{
+    Unset,
+    OtherError,
+    ConnectionReset,
+    FlowControlWindowExceeded,
+    KeepAliveTimeout,
+    InsufficientTlsVersion,
+    InvalidHandshake,
```

Comment:
This seems similar to UnexpectedFrame

Reply:
What do you mean?

InvalidHandshake is only used for HTTP/2 and occurs when the client doesn't start the connection with the HTTP/2 preface.

Reply:
I think that comment was supposed to be about `FrameAfterStreamClose`.  My bad.

Reply:
Hm, I wonder if I was commenting on an older commit.  I can't find a refresh button in the VS Code UI that seems to cover moving forward to the latest revision of the PR.